tree/tree/src/TTree.cxx

```diff
@@ -4383,53 +4383,56 @@ Int_t TTree::Fill()
 {
    Int_t nbytes = 0;
    Int_t nerror = 0;
-   Int_t nb = fBranches.GetEntriesFast();
-   if (nb == 1) {
-      // Case of one single super branch. Automatically update
-      // all the branch addresses if a new object was created.
-      TBranch* branch = (TBranch*) fBranches.UncheckedAt(0);
-      branch->UpdateAddress();
-   }
-   if (fBranchRef) {
+   Int_t nbranches = fBranches.GetEntriesFast();
+
+   // Case of one single super branch. Automatically update
+   // all the branch addresses if a new object was created.
+   if (nbranches == 1)
+      ((TBranch*)fBranches.UncheckedAt(0))->UpdateAddress();
+
+   if (fBranchRef)
       fBranchRef->Clear();
-   }
 
+#ifdef R__USE_IMT
    ROOT::Internal::TBranchIMTHelper imtHelper;
-   #ifdef R__USE_IMT
+
    if (fIMTEnabled) {
       fIMTFlush = true;
       fIMTZipBytes.store(0);
       fIMTTotBytes.store(0);
    }
-   #endif
+#endif
 
-   for (Int_t i = 0; i < nb; ++i) {
+   for (Int_t i = 0; i < nbranches; ++i) {
       // Loop over all branches, filling and accumulating bytes written and error counts.
       TBranch* branch = (TBranch*) fBranches.UncheckedAt(i);
-      if (branch->TestBit(kDoNotProcess)) {
+
+      if (branch->TestBit(kDoNotProcess))
          continue;
-      }
+
+#ifndef R__USE_IMT
+      Int_t nwrite = branch->FillImpl(nullptr);
+#else
       Int_t nwrite = branch->FillImpl(fIMTEnabled ? &imtHelper : nullptr);
-      if (nwrite < 0)  {
-         if (nerror < 2) {
-            Error("Fill", "Failed filling branch:%s.%s, nbytes=%d, entry=%lld\n"
-                  " This error is symptomatic of a Tree created as a memory-resident Tree\n"
-                  " Instead of doing:\n"
-                  "    TTree *T = new TTree(...)\n"
-                  "    TFile *f = new TFile(...)\n"
-                  " you should do:\n"
-                  "    TFile *f = new TFile(...)\n"
-                  "    TTree *T = new TTree(...)",
-                  GetName(), branch->GetName(), nwrite,fEntries+1);
-         } else {
-            Error("Fill", "Failed filling branch:%s.%s, nbytes=%d, entry=%lld", GetName(), branch->GetName(), nwrite,fEntries+1);
-         }
-         ++nerror;
-      } else {
+#endif
+
+      if (nwrite > 0) {
          nbytes += nwrite;
+      } else {
+         ++nerror;
+         Error("Fill", "Failed filling branch:%s.%s, nbytes=%d, entry=%lld\n"
```

Comment:
Shouldn't errors be prominent in the output? What if the user is printing something and filling? They will likely miss the problem entirely.